### PR TITLE
add paddle job resource support when using extensiontemplate

### PR DESCRIPTION
**Pull request by tornado404**
### PR types
Function optimization

### PR changes
Others

### Describe
add paddle job resource support when using extensiontemplate

### CODE REVIEW

**Review by luoyuedong — APPROVED** [review]

File: pkg/job/runtime_v2/job/paddle/kube_paddle_job.go [hunk 1]
```diff
@@ -382,3 +392,75 @@ func (pj *KubePaddleJob) getJobStatus(jobStatus *paddlejobv1.PaddleJobStatus) (p
 	}
 	return status, msg, nil
 }
+
+func (pj *KubePaddleJob) patchResource(pdj *paddlejobv1.PaddleJob, job *api.PFJob) error {
+	log.Infof("patch resource in paddlejob")
+	if len(job.Tasks) == 0 {
+		log.Debugf("no resources to be configured")
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/job/paddle/kube_paddle_job.go:399]: Missing return?

File: pkg/job/runtime_v2/job/paddle/kube_paddle_job.go [hunk 2]
```diff
@@ -106,13 +111,18 @@ func (pj *KubePaddleJob) Submit(ctx context.Context, job *api.PFJob) error {
...
 func (pj *KubePaddleJob) customPaddleJob(pdj *paddlejobv1.PaddleJob, job *api.PFJob) error {
+	log.Infof("customPaddleJob fill resource")
 	if pdj == nil || job == nil {
 		return fmt.Errorf("jobSpec or PFJob is nil")
 	}
 	if err := pj.validateCustomYaml(pdj, job.ID); err != nil {
 		log.Errorf("validate custom yaml for %s failed, err: %v", pj.String(job.ID), err)
 		return err
 	}
+	if err := pj.patchResource(pdj, job); err != nil {
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/job/paddle/kube_paddle_job.go:122]: Move `patchResource` to line `117`.

File: pkg/apiserver/controller/job/create.go [hunk 3]
```diff
@@ -116,6 +135,59 @@ func validateJob(ctx *logger.RequestContext, request *CreateJobInfo) error {
...
+	var err error
+	sumResource := resources.EmptyResource()
+	for index, member := range request.Members {
+		member.Flavour, err = flavour.GetFlavourWithCheck(member.Flavour)
+		if err != nil {
+			log.Errorf("get flavour failed, err:%v", err)
+			return err
+		}
+		request.Members[index].Flavour.ResourceInfo = member.Flavour.ResourceInfo
+		// sum = sum + member.Replicas * member.Flavour.ResourceInfo
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/create.go]: Remove comment.

File: pkg/job/runtime_v2/job/paddle/kube_paddle_job.go (hunk 1 above)
Comment (tornado404) [reply pkg/job/runtime_v2/job/paddle/kube_paddle_job.go:399]: done

File: pkg/job/runtime_v2/job/paddle/kube_paddle_job.go (hunk 2 above)
Comment (tornado404) [reply pkg/job/runtime_v2/job/paddle/kube_paddle_job.go:122]: done

**Review by D0m021ng — APPROVED** [review]

**Pull request merged**